pkg/config/config.go

```diff
@@ -56,7 +56,12 @@ func FromYAML(contents []byte) (*Config, error) {
 		return nil, fmt.Errorf("Failed to parse config yaml: %w", err)
 	}
 	// Everything assumes Build is not nil
-	if config.Build == nil {
+	if string(contents) != "" && config.Build != nil {
+		err := Validate(string(contents), "")
+		if err != nil {
+			return nil, err
+		}
+	} else {
```

Comment:
Should this be inside this conditional? Wouldn't it make sense to always validate the config, even when `build` is not set?

Reply:
This was added to make sure the `validate` functionality is backward compatible.  Right now `cog.yaml` is not required and also build is not a required keyword. I think this is actually nice and let's user avoid writing a cog.yaml file if they don't want to. This can definitely be changes in future and restrictions can be added.

Let me know what do you think.

Reply:
Gotchya. Sounds sensible to keep current behavior.